docs/guides/auth.md

```diff
@@ -149,50 +152,159 @@ These authentication mechanisms will be available in all gRPC's supported langua
 described above appear in each language: more languages are coming soon.
 
-###SSL/TLS for server authentication and encryption (Ruby)
+### Ruby
 
+#### Base case - no encryption or authentication
 ```ruby
-# Base case - No encryption
+
 stub = Helloworld::Greeter::Stub.new('localhost:50051', :this_channel_is_insecure)
 ...
+```
 
-# With server authentication SSL/TLS
+#### With server authentication SSL/TLS
+
+```ruby
 creds = GRPC::Core::Credentials.new(load_certs)  # load_certs typically loads a CA roots file
 stub = Helloworld::Greeter::Stub.new('localhost:50051', creds)
 ```
 
-###SSL/TLS for server authentication and encryption (C#)
+#### Authenticate with Google using scopeless credentials (recommended approach)
+
+```ruby
+require 'googleauth'  # from http://www.rubydoc.info/gems/googleauth/0.1.0
+...
+ssl_creds = GRPC::Core::ChannelCredentials.new(load_certs)  # load_certs typically loads a CA roots file
+authentication = Google::Auth.get_application_default()
+call_creds = GRPC::Core::CallCredentials.new(authentication.updater_proc)
+combined_creds = ssl_creds.compose(call_creds)
+stub = Helloworld::Greeter::Stub.new('greeter.googleapis.com', combined_creds)
+```
+
+#### Authenticate with Google using Oauth2 token (legacy approach)
+
+```ruby
+require 'googleauth'  # from http://www.rubydoc.info/gems/googleauth/0.1.0
+...
+ssl_creds = GRPC::Core::ChannelCredentials.new(load_certs)  # load_certs typically loads a CA roots file
+scope = 'https://www.googleapis.com/auth/grpc-testing'
+authentication = Google::Auth.get_application_default(scope)
+call_creds = GRPC::Core::CallCredentials.new(authentication.updater_proc)
+combined_creds = ssl_creds.compose(call_creds)
+stub = Helloworld::Greeter::Stub.new('greeter.googleapis.com', combined_creds)
+```
+
+### C&#35;
+
+#### Base case - no encryption or authentication
 
 ```csharp
-// Base case - No encryption/authentication
 var channel = new Channel("localhost:50051", ChannelCredentials.Insecure);
 var client = new Greeter.GreeterClient(channel);
 ...
+```
 
-// With server authentication SSL/TLS
+#### With server authentication SSL/TLS
+
+```csharp
 var channelCredentials = new SslCredentials(File.ReadAllText("roots.pem"));  // Load a custom roots file.
 var channel = new Channel("myservice.example.com", channelCredentials);
 var client = new Greeter.GreeterClient(channel);
 ```
 
-###SSL/TLS for server authentication and encryption (Python)
+####Authenticate with Google using scopeless credentials (recommended approach)
+```csharp
+using Grpc.Auth;  // from Grpc.Auth NuGet package
+...
+// Loads Google Application Default Credentials with publicly trusted roots.
+var channelCredentials = await GoogleGrpcCredentials.GetApplicationDefaultAsync();
+
+var channel = new Channel("greeter.googleapis.com", channelCredentials);
+var client = new Greeter.GreeterClient(channel);
+...
+```
+
+####Authenticate with Google using OAuth2 token (legacy approach)
```

Comment:
Can we keep this for a little longer? We've recently had some issues with JWT access to some google APIs and also not all languagues support JWT fully yet.

Also, I'd prefer if we removed the OAuth2 sections for all languages in a separate PR (with the description of the PR stating why we are removing these).